iStore/Location/LocationDataManager.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  LocationViewModel.swift
+//  iStore
+//
+//  Created by Mohamed Sayed on 12.04.24.
+//
+
+import Foundation
+import CoreLocation
+
+final class LocationDataManager: NSObject, ObservableObject {
+    private let locationManager = CLLocationManager()
+    
+    @Published var address: String
+    @Published var postalCode: String
+    @Published var city: String
+    @Published var state: String
+    @Published var latitude: Double
+    @Published var longitude: Double
+    
+    @Published var isCurrentLocation = false
+    
+    init(address: String, postalCode: String, city: String, state: String, latitude: Double, longitude: Double) {
+        self.address = address
+        self.postalCode = postalCode
+        self.city = city
+        self.state = state
+        self.latitude = latitude
+        self.longitude = longitude
+    }
+    
+    func requestUserLocationPermission() {
+        locationManager.delegate = self
+        
+        switch locationManager.authorizationStatus {
+        case .authorizedWhenInUse:
+            locationManager.requestLocation()
+            
+        case .restricted:
+            break
+            
+        case .denied:
+            break
+            
+        case .notDetermined:
+            locationManager.requestWhenInUseAuthorization()
+            
+        default:
+            break
+        }
+        
+        locationManager.startUpdatingLocation()
+    }
+    
+    func getUserAddress(completion: @escaping (CLPlacemark) -> Void) {
+        if let Location = self.locationManager.location {
```

Comment:
Why `if let` here not `guard`?

Reply:
I think both are applicable.
but guard let is better as It returns earlier if there is an error.